### ISSUE TITLE
Update Text Field Values Generates Text Behind The Text Field

### ISSUE THREAD

**Issue by AhmedAlZabidi** [py-pdf/pypdf#1618]
1- We created a simple empty word document and then convert it to a pdf
2- We prepared a form by adding a text field (we named it 'due_date') to the pdf file
3- We saved the pdf file
3- We run the below script to generate a new filled pdf file
4- We opened the newly generated file and **we could not see the updated values**. Note: when we click on the text field, the updated value appears. Also, when we open the file in Google chrome browser, the updated values will be shown

## Environment
Python 3.8.10

## Code + PDF

```python
from pypdf import PdfReader, PdfWriter

reader = PdfReader("test.pdf")
writer = PdfWriter()

page = reader.pages[0]
fields = reader.get_fields()

writer.add_page(page)

writer.update_page_form_field_values(
    writer.pages[0], {
        'due_date': 'hello'
    }
)

# write "output" to PyPDF2-output.pdf
with open("filled-out.pdf", "wb") as output_stream:
    writer.write(output_stream)
```
The input file:
[test.pdf](https://github.com/py-pdf/pypdf/files/10688486/test.pdf)

The output file:

[filled-out.pdf](https://github.com/py-pdf/pypdf/files/10688480/filled-out.pdf)

**Comment by 3sticks** [comment]
Having this same problem myself. Added text will not show up until clicking on the text field, unless opened in a browser.

**Comment by csears123** [comment]
This is related to issue #355

**Comment by pubpub-zz** [comment]
I've been able to get the text displayed using append() (https://pypdf.readthedocs.io/en/stable/user/merging-pdfs.html#add-page-insert-page)
it is copying the fields tree whereas append_page doesn't:
```
from pypdf import PdfWriter

writer = PdfWriter()
writer.append("test.pdf")

writer.update_page_form_field_values(
    writer.pages[0], {
        'due_date': 'hello'
    }
)
writer.write("filled-out.pdf")
```

This issue should be closed

**Comment by pubpub-zz** [comment]
this issue is closed with #1864.
